Source/WTF/wtf/RunLoop.cpp

```diff
@@ -170,19 +170,6 @@ void RunLoop::dispatch(Function<void()>&& function)
         wakeUp();
 }
 
-Ref<RunLoop::DispatchTimer> RunLoop::dispatchAfter(Seconds delay, Function<void()>&& function)
-{
-    RELEASE_ASSERT(function);
-    Ref<DispatchTimer> timer = adoptRef(*new DispatchTimer(*this));
-    timer->setFunction([timer = timer.copyRef(), function = WTFMove(function)]() mutable {
-        Ref<DispatchTimer> protectedTimer { WTFMove(timer) };
-        function();
-        protectedTimer->stop();
-    });
```

Comment:
How about making `DispatchTimer` as `ThreadSafeWeakPtr` in this function, and checking whether it is not destroyed before calling calling `function`?

Reply:
If we end up here to convert the weak ptr to strong ptr, we already have accessed the destroyed thing.

Effectively:
```
timer->setMyKeepalive(ThreadSafeWeakPtr { timer });

...

auto protectedTimer = timer->m_myKeepAlive.strongRef();
```

which does not make sense.

> checking whether it is not destroyed before

In other words: the "it" is the `timer` which is being invoked via DispatchTimer::fired() of the `timer` instance. It does not make sense to be promoting weak `timer` to strong one, as we are already accessing the `timer`.

(ThreadSafe WeakPtr support makes the refcount a bit more inefficient.)

Reply:
I think the issue here is that function is holding a strong reference to the timer, and timer holds a strong reference to the function, cyclic reference.
Thus, the usual fix is making one side weak. In this case, timer is owning the function. Thus, timer should hold function strongly and function should hold timer weakly.
So, my proposed change for this fix is,
```
    timer->setFunction([weakTimer = ThreadSafeWeakPtr { timer }, function = WTFMove(function)]() mutable {
        if (auto timer = weakTimer.get()) {
            function();
            timer->stop();
        }
    });
```

> (ThreadSafe WeakPtr support makes the refcount a bit more inefficient.)

New implementation is allocating Objective-C Block additionally. And I wonder if this is more inefficient.

Reply:
Or, how about,

1. As your patch does, limit the interface and make this DelayedDispatch
2. And function holds the ref strongly
3. And `invalidate` function will clear `m_function` in DelayedDispatch, breaking ref from `DelayedDispatch` to the function.
4. And remove setFunction feature.